ArcGISRuntimeSDKQt_SampleViewers/ArcGISRuntimeSDKQt_CppSamples/ZipHelper.cpp

```diff
@@ -0,0 +1,715 @@
+// [Legal]
+// Copyright 2024 Esri.
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// http://www.apache.org/licenses/LICENSE-2.0
+
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// [Legal]
+/// \file ZipHelper.cpp
+
+#include "pch.hpp" // IWYU pragma: keep
+#include "ZipHelper.h"
+
+// Qt headers
+#include <QFile>
+
+// enable logging
+//#define ZIP_LOGGING_ENABLED
```

Comment:
The original file had optimized category logging which carried a lot of baggage and overhead to move into the sample viewer. Rather than trying to keep that in and working, it was simpler to replace them with qDebug() and just hide them under a define if logging is needed for debugging.